src/plugins/dialogue_box/DialogueBox.tsx

```diff
@@ -0,0 +1,129 @@
+import React, {useState} from 'react';
+import {
+    State, 
+    IAnnotation, 
+    ISinglePack, 
+    transformBackAnnotation,
+    transformPack,
+} from "../../nlpviewer"
+import {PluginComponentProp} from "../lib/interface"
+import style from "./DialogueBox.module.css"
+import TextInput from "./TextInput"
+import { useParams } from 'react-router-dom';
+import {
+    editText,
+    addAnnotation,
+    loadNlpModel,
+    runNlp,
+} from '../../app/lib/api';
+
+function Utterance(text: string, annotation: IAnnotation){
+    if (annotation.attributes.speaker === 'ai'){
+        return <div className={style.bubble_container} key={'utterance_container_' + annotation.id}> 
+            <div className={style.bubble_left} key={'utterance_bubble_' + annotation.id}>    
+                <div key={'utterance_' + annotation.id}>
+                    {text.substring(annotation.span.begin, annotation.span.end)}
+                </div>
+            </div>
+        </div>
+    }else if (annotation.attributes.speaker === 'user'){
+        return <div className={style.bubble_container} key={'utterance_container_' + annotation.id}>        
+            <div className={style.bubble_right} key={'utterance_bubble_' + annotation.id}>    
+                <div key={'utterance_' + annotation.id}>
+                    {text.substring(annotation.span.begin, annotation.span.end)}
+                </div>
+            </div>
+        </div>
+    } else{
+        return  <div className={style.bubble_container} key={'utterance_container_' + annotation.id}>        
+        <div className={style.bubble_right} key={'utterance_bubble_' + annotation.id}>    
+            ...
+        </div>
+        </div>
+    }
+}
+
+function DialogueBox(props: PluginComponentProp) {
+    let { id } = useParams();
+
+    // Avoid confusion between doc_id and annotation id.
+    const doc_id = id;
+    
+    const [pack, setPack] = useState<ISinglePack | null> (props.appState.textPack);
+
+    if(!pack){
+        return null;
+    }
+
+    let {annotations, text} = pack;
+
+    const utterances = annotations.filter(
+        (ann) => {
+            // TODO: This did not read the class hierarchy.
+            return ann.legendId === "ft.onto.base_ontology.Utterance"
+        }
+    );
+
+    var model_ok = false
+    const model_name = 'content_rewriter'
+
+    // Call API to load the NLP model of name "model_name".
+    loadNlpModel(model_name).then(() =>{
+        model_ok = true
+    });
```

Comment:
better to call api in the useEffect hook, otherwise it will be called every time there is is re-render

```js
useEffect(() => {
  loadNlpModel(model_name).then(() =>{
    model_ok = true
  });
}, [])
```

Reply:
done